SharpSource/SharpSource.Test/EnumWithoutDefaultValueTests.cs

```diff
@@ -81,6 +81,33 @@ public async Task EnumWithoutDefaultValue_RightName_RightValueDuplicated(string
 enum Test {{
     A = 0,
     {memberName} = 0,
+}}";
+
+        await VerifyCS.VerifyNoDiagnostic(original);
+    }
+
+    [TestMethod]
+    [DataRow("None")]
+    [DataRow("Unknown")]
+    public async Task EnumWithoutDefaultValue_RightName_NegativeValue(string memberName)
+    {
+        var original = $@"
+enum [|Test|] : short {{
+    {memberName} = -1
+}}";
+
+        await VerifyCS.VerifyNoDiagnostic(original);
+    }
+
+    [TestMethod]
+    [DataRow("None")]
+    [DataRow("Unknown")]
+    public async Task EnumWithoutDefaultValue_RightName_WithDifferentType(string memberName)
```

Comment:
Can you make this a parameterised test that has a `[DataRow]` for each type an enum can take? That way this unit test is exhaustive of the different combinations. I believe that means `byte`, `sbyte`, `short`, `ushort`, `int`, `uint`, `long`, and `ulong`.

I think you can use a constant `memberName` here to simplify the test

Reply:
Sorry for the delay. Had to get home from work first.

Only just now saw your comment about simplifying it. I'll let it be up to you if you still want it. :)